### PR TITLE
Added default values as class variables

### PR DESCRIPTION
**Pull request by esteve**
This adds the default values (if any) as constants (class variables)

Connects to #67

### CODE REVIEW

**Comment by esteve** [comment]
CI:

http://ci.ros2.org/job/ci_linux/846/
http://ci.ros2.org/job/ci_osx/709/
http://ci.ros2.org/job/ci_windows/920/

**Comment by dirk-thomas** [comment]
The variable can't be called `DEFAULT_@(field.name)` since this would collide with a constant which starts with `DEFAULT_`.

**Comment by esteve** [comment]
@dirk-thomas what do you think would be a good sane name?

**Comment by wjwwood** [comment]
Field names and constant names cannot contain double underscore nor can they start with an underscore, so either `@(field.name)__default` or `_default_@(field.name)`?

**Comment by esteve** [comment]
@wjwwood `@(field.name)__default` looks good to me

**Comment by esteve** [comment]
CI jobs with the new names:

http://ci.ros2.org/job/ci_linux/848/
http://ci.ros2.org/job/ci_osx/711/
http://ci.ros2.org/job/ci_windows/920/

File: rosidl_generator_py/resource/_msg.py.template [hunk 1]
```diff
@@ -19,6 +28,11 @@ class @(spec.base_type.type)__Constants(type):
 @[for constant in spec.constants]@
             '@(constant.name)': @constant_value_to_py(constant.type, constant.value),
 @[end for]@
+@[for field in spec.fields]@
+@[if field.default_value]@
+            '@(field.name)__default': @value_to_py(field.type, field.default_value),
```
Comment (dirk-thomas) [inline rosidl_generator_py/resource/_msg.py.template]: PEP8: constants must be uppercase.

Comment (esteve) [inline rosidl_generator_py/resource/_msg.py.template]: So, would you be ok with `@(field.name.upper())__DEFAULT` ?

Comment (esteve) [inline rosidl_generator_py/resource/_msg.py.template]: I've updated the branch to use `@(field.name.upper())__DEFAULT` and also that arrays that have default values are turned into `tuple`s so they are immutable.

**Comment by esteve** [comment]
Moar CI:

http://ci.ros2.org/job/ci_linux/849/
http://ci.ros2.org/job/ci_osx/712/
http://ci.ros2.org/job/ci_windows/920/

**Comment by dirk-thomas** [comment]
The constructor should use the default value constant.

**Comment by esteve** [comment]
@dirk-thomas done. Do you have more feedback before I trigger the CI jobs?

**Comment by dirk-thomas** [comment]
Currently I can't think of anything else.

**Comment by esteve** [comment]
CI:

http://ci.ros2.org/job/ci_linux/852/
http://ci.ros2.org/job/ci_osx/715/
http://ci.ros2.org/job/ci_windows/923/

**Comment by esteve** [comment]
I've addressed all feedback and the tests pass, this should be mergeable.

**Comment by dirk-thomas** [comment]
lgtm

**Pull request merged**